docs/csharp/csharp-6.md

```diff
@@ -1,17 +1,570 @@
+the specifically address those interactions. The goal is to
+minimize any chances of breaking changes in existing codebases,
+including yours.
+
+Extension methods are only in scope when called using the
+extension method invocation syntax, not when called as a static method.
+You'll often see this in LINQ queries. You can import the LINQ pattern
+by importing `System.Linq.Enumerable`. 
+
+[!code-csharp[UsingStaticLinq](../../samples/snippets/csharp/new-in-6/newcode.cs#usingStaticLinq)]
+
+This imports all the methods in the `System.Linq.Enumerable` class.
+However, the extension methods are only in scope when called as extension
+methods. They are not in scope if they are called as though they are static
+methods:
+
+[!code-csharp[UsingStaticLinqMethod](../../samples/snippets/csharp/new-in-6/newcode.cs#UsingStaticLinkMethod)]
+
+This decision is because extension methods are typically called using
+extension method invocation expressions. In the rare case where they are
+called using the static method call syntax it is to resolve ambiguity.
+Requiring the class name as part of the invocation seems wise.
+
+There's one last feature of `static using`. The `static using` directive
+also imports any nested types. That enables you to reference any nested
+types without qualification.
+
+## Null-conditional operators
+
+Null values complicate code. You need to check every access
+of variables to ensure you are not dereferencing `null`. The
+***null conditional operator*** makes those checks much easier
+and fluid.
+
+Simply replace the member access `.` with `?.`:
+
+[!code-csharp[NullConditional](../../samples/snippets/csharp/new-in-6/program.cs#NullConditional)]
+
+In the above example, the variable `first` is assigned `null` if the person object
+is null. Otherwise, it gets assigned the value of the `FirstName` property. Most importantly,
+the `?.` means that the above line of code does not generate a `NullReferenceException` when
+the `person` variable is null. Instead, it short-circuits and produces null.
+
+Also, note that this expression returns a `string`, regardless of the value of `person`.
+In the case of short circuiting, the null value returned is typed to match the full
+expression.
+
+You can often use this construct with the ***null coalescing*** operator to assign
+default values when one of the properties are null:
+
+[!code-csharp[NullCoalescing](../../samples/snippets/csharp/new-in-6/program.cs#NullCoalescing)]
+
+The right hand side operand of the `?.` operator is not limited to properties or fields.
+You can also use it to conditionally invoke methods. The most common use of member functions
+ with the null conditional operator is to safely invoke delegates
+(or event handlers) that may be null.  You'll do this by calling the delegate's `Invoke` method
+using the `?.` operator to access the member. You can see an example where we discuss 
+[delegate patterns](delegates-patterns.md#Handling-Null-Delegates).
```

Comment:
anchor needs to be lowercase: handling-null-delegates or it doesn't work.

I also think that the expression "where we discuss" seems a bit strange. Maybe change this to:
You can see an example on [delegate patterns](delegates-patterns.md#handling-null-delegates) topic.